src/EFCore.PG/Storage/Internal/Mapping/NpgsqlTsQueryTypeMapping.cs

```diff
@@ -0,0 +1,24 @@
+﻿using System.Text;
+using Microsoft.EntityFrameworkCore.Utilities;
+using NpgsqlTypes;
+
+namespace Microsoft.EntityFrameworkCore.Storage.Internal.Mapping
+{
+    public class NpgsqlTsQueryTypeMapping : NpgsqlTypeMapping
+    {
+        public NpgsqlTsQueryTypeMapping() : base("tsquery", typeof(NpgsqlTsQuery), NpgsqlDbType.TsQuery) { }
+
+        protected override string GenerateNonNullSqlLiteral(object value)
+        {
+            Check.NotNull(value, nameof(value));
+            var query = (NpgsqlTsQuery)value;
+            var builder = new StringBuilder();
+            builder.Append(" ");
+            query.Write(builder, true);
+            builder.Replace("'", "''");
+            builder[0] = '\'';
+            builder.Append("'::tsquery");
```

Comment:
The literal representations as usually of the form `HSTORE 'blablabla'` rather than `'blablabla'::hstore`, it's probably a good idea to align to that.

Reply:
Hmmm I didn't know this syntax. I will update it to match.